Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreImpl.cpp

```diff
@@ -61,22 +66,35 @@ String BackgroundFetchStoreImpl::getFilename(const ServiceWorkerRegistrationKey&
     return iterator->value.fetchToFilenames.get(std::make_pair(key.scope().string(), identifier));
 }
 
-void BackgroundFetchStoreImpl::initializeFetches(BackgroundFetchEngine& engine, const ServiceWorkerRegistrationKey& key, CompletionHandler<void()>&& callback)
+void BackgroundFetchStoreImpl::registerFetch(const ClientOrigin& origin, const ServiceWorkerRegistrationKey& key, const String& backgroundFetchIdentifier, String&& fetchStorageIdentifier)
```

Comment:
An unregisterFetch method is clearer than doing the work in clearFetch/clearAllFetches.
I'll update the patch accordingly.

Reply:
An unregisterFetch is also less efficient, since for clearFetch/clearAllFetches, we are already doing the find.
Also, m_perClientOriginFetches is growing so that, if we initialize once, we do not try again to reinitialize.